assert/assertions.go

```diff
@@ -246,7 +248,7 @@ func Fail(t TestingT, failureMessage string, msgAndArgs ...interface{}) bool {
 
 // Implements asserts that an object is implemented by the specified interface.
 //
-//    assert.Implements(t, (*MyInterface)(nil), new(MyObject), "MyObject")
```

Comment:
We cannot do it for this version, since it would be a breaking change, but I'm thinking that testify's assertions provide plenty of context around the failure.

I a v2 of testify, we would probably have assertions not take any message at all, and offer the "format" assertions like `Errorf` to those who want to provide some context.

This has some extra benefits, since people right now make mistakes like this:

```go
assert.Error(t, err, "expected error message")
```

Because they believe that would check that `err.Error() == "expected error message"` and the code compiles and runs.